crates/builder/src/tasks/block.rs

```diff
@@ -76,13 +79,60 @@ impl InProgressBlock {
     }
 }
 
+/// Implements a poller for the block builder to pull transactions from the transaction pool.
+pub struct TxPoller {
+    pub config: BuilderConfig,
+    pub client: Client,
+}
+
+impl TxPoller {
+    // returns a new TxPoller with the given config
+    pub fn new(config: &BuilderConfig) -> Self {
+        Self { config: config.clone(), client: Client::new() }
+    }
+
+    // polls the transaction pool for transactions.
+    pub async fn poll_transactions(&self) -> Result<Vec<TxEnvelope>, Error> {
+        let result = self.client.get(self.config.tx_pool_url.to_string() + "/get").send().await?;
+
+        let mut transactions: Vec<TxEnvelope> = Vec::new();
+
+        // OPTIMIZE - This could be made cleaner with a method chain approach that I don't have time to think up right now
+        //
+        // Parse the response as a JSON array of key value pairs
+        let parsed: Value = from_str(&result.text().await?)?;
```

Comment:
what is the current format returned in json? usually the way we'd do this is skip `Value` and go straight to a struct

currently you're getting data of this format:

```
[ 
    { 
        "value": "{\"to\":\"\"}"
    },
    { 
    },
   ...
]
```

note the inner stringified JSON in JSON, which is probably not what you want.

I think that you actually want is something like this:

`let txns: Vec<TxEnvelope> = serde_json::from_str(&result.text().await?)?;`

which would parse json like:

```
[
    { 
        "to: "0x...",
        "data": "0x..."
    },
    ....
]
```

Reply:
The format from the SDK comes back as an array of objects with fields key and value, which correspond to the transaction hash and transaction envelope, like so:

```
[{
"key": "txn_hash",
"value": "{"to": "0x00..." .... },
}, ... ]
```

I tried slimming it down and parsing it like you suggested but that didn't work, and my attempts to fiddle with it didn't yield anything better than this.

Reply:
i don't love that it comes out that way but here's the better version:

```
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TxPoolResponse {
    #[serde(rename = "key")]
    put hash: B256,
    #[serde(rename = "value")]
    pub tx: TxEnvelope,
}

let responses: Vec<TxPoolResponse> = serde_json::from_str(&text)
```

Reply:
This refactor is enabled in [tx-pool PR #8](https://github.com/init4tech/tx-pool-webservice/pull/8).